--- a/controller.pp	2015-04-01 14:50:14.932890426 +0000
+++ b/controller.pp	2015-04-01 14:52:02.688415282 +0000
@@ -656,16 +656,18 @@
       Service<| title == 'neutron-server' |> ->
         Openstack::Network::Create_router <||>
 
-      openstack::network::create_network{'net04':
-        netdata => $nets['net04']
-      } ->
-      openstack::network::create_network{'net04_ext':
-        netdata => $nets['net04_ext']
-      } ->
-      openstack::network::create_router{'router04':
-        internal_network => 'net04',
-        external_network => 'net04_ext',
-        tenant_name      => $keystone_admin_tenant
+      if ! empty($nets) {
+        openstack::network::create_network{'net04':
+          netdata => $nets['net04']
+        } ->
+        openstack::network::create_network{'net04_ext':
+          netdata => $nets['net04_ext']
+        } ->
+        openstack::network::create_router{'router04':
+          internal_network => 'net04',
+          external_network => 'net04_ext',
+          tenant_name      => $keystone_admin_tenant
+        }
       }
     }
     $pnets = $neutron_settings['L2']['phys_nets']
