diff --git a/nova/objects/fixed_ip.py b/nova/objects/fixed_ip.py
index 2482b63..97d4a1a 100644
--- a/nova/objects/fixed_ip.py
+++ b/nova/objects/fixed_ip.py
@@ -61,6 +61,10 @@ class FixedIP(obj_base.NovaPersistentObject, obj_base.NovaObject):
             self.instance.obj_make_compatible(
                     primitive['instance']['nova_object.data'], '1.13')
             primitive['instance']['nova_object.version'] = '1.13'
+        if target_version < (1, 2) and 'network' in primitive:
+            self.network.obj_make_compatible(
+                    primitive['network']['nova_object.data'], '1.1')
+            primitive['network']['nova_object.version'] = '1.1'
 
     @property
     def floating_ips(self):
diff --git a/nova/objects/service.py b/nova/objects/service.py
index 693755e..8de251d 100644
--- a/nova/objects/service.py
+++ b/nova/objects/service.py
@@ -49,8 +49,8 @@ class Service(base.NovaPersistentObject, base.NovaObject):
         target_version = utils.convert_version_to_tuple(target_version)
         if target_version < (1, 3) and 'compute_node' in primitive:
             self.compute_node.obj_make_compatible(
-                    primitive['compute_node']['nova_object.data'], '1.4')
-            primitive['compute_node']['nova_object.version'] = '1.4'
+                    primitive['compute_node']['nova_object.data'], '1.3')
+            primitive['compute_node']['nova_object.version'] = '1.3'
 
     @staticmethod
     def _do_compute_node(context, service, db_service):
