diff --git a/tasks_serializer.py  b/tasks_serializer.py
index cee16fa..d90e91d 100644
--- tasks_serializer.py
+++ tasks_serializer.py
@@ -225,9 +225,7 @@ class CopyKeys(GenericRolesHook):
         for file_path in self.task['parameters']['files']:
             file_path['src'] = file_path['src'].format(
                 CLUSTER_ID=self.cluster.id)
-        uids = self.get_uids()
-        yield templates.make_generic_task(
-            uids, self.task)
+        return super(CopyKeys, self).serialize()
