diff -Nur neutron/neutron.conf /etc/neutron/neutron.conf
--- neutron/neutron.conf	2014-12-16 15:10:58.000000000 +0000
+++ /etc/neutron/neutron.conf	2015-03-12 11:20:08.369824447 +0000
@@ -1,6 +1,6 @@
 [DEFAULT]
 # Print more verbose output (set logging level to INFO instead of default WARNING level).
-# verbose = False
+verbose = True

 # =========Start Global Config Option for Distributed L3 Router===============
 # Setting the "router_distributed" flag to "True" will default to the creation
@@ -14,6 +14,7 @@

 # Print debugging output (set logging level to DEBUG instead of default WARNING level).
 # debug = False
+debug = True

 # Where to store Neutron state files.  This directory must be writable by the
 # user executing the agent.
@@ -32,20 +33,20 @@
 # (not user_stderr) and (not log_file) -> stdout
 # publish_errors                       -> notification system

-# use_syslog = False
-# syslog_log_facility = LOG_USER
+use_syslog = True
+syslog_log_facility = LOG_LOCAL4

 # use_stderr = True
-# log_file =
-# log_dir =
+log_file =/var/log/neutron/server.log
+log_dir =/var/log/neutron

-# publish_errors = False
+publish_errors = False

 # Address to bind the API server to
-# bind_host = 0.0.0.0
+bind_host = 0.0.0.0

 # Port the bind the API server to
-# bind_port = 9696
+bind_port = 9696

 # Path to the extensions.  Note that this can be a colon-separated list of
 # paths.  For example:
@@ -60,7 +61,7 @@
 # previous versions, the class name of a plugin can be specified instead of its
 # entrypoint name.
 #
-core_plugin = ml2
+core_plugin = neutron.plugins.ml2.plugin.Ml2Plugin
 # Example: core_plugin = ml2

 # (ListOpt) List of service plugin entrypoints to be loaded from the
@@ -71,19 +72,20 @@
 #
 # service_plugins =
 # Example: service_plugins = router,firewall,lbaas,vpnaas,metering
+service_plugins =neutron.services.l3_router.l3_router_plugin.L3RouterPlugin,neutron.services.firewall.fwaas_plugin.FirewallPlugin,neutron.services.metering.metering_plugin.MeteringPlugin

 # Paste configuration file
 # api_paste_config = api-paste.ini

 # The strategy to be used for auth.
 # Supported values are 'keystone'(default), 'noauth'.
-# auth_strategy = keystone
+auth_strategy = keystone

 # Base MAC address. The first 3 octets will remain unchanged. If the
 # 4h octet is not 00, it will also be used. The others will be
 # randomly generated.
 # 3 octet
-# base_mac = fa:16:3e:00:00:00
+base_mac = fa:16:3e:00:00:00
 # 4 octet
 # base_mac = fa:16:3e:4f:00:00

@@ -97,16 +99,17 @@

 # Maximum amount of retries to generate a unique MAC address
 # mac_generation_retries = 16
+mac_generation_retries = 32

 # DHCP Lease duration (in seconds).  Use -1 to
 # tell dnsmasq to use infinite lease times.
-# dhcp_lease_duration = 86400
+dhcp_lease_duration = 120

 # Allow sending resource operation notification to DHCP agent
-# dhcp_agent_notification = True
+dhcp_agent_notification = True

 # Enable or disable bulk create/update/delete operations
-# allow_bulk = True
+allow_bulk = True
 # Enable or disable pagination
 # allow_pagination = False
 # Enable or disable sorting
@@ -114,7 +117,7 @@
 # Enable or disable overlapping IPs for subnets
 # Attention: the following parameter MUST be set to False if Neutron is
 # being used in conjunction with nova security groups
-# allow_overlapping_ips = False
+allow_overlapping_ips = True
 # Ensure that configured gateway is on subnet. For IPv6, validate only if
 # gateway is not a link local address. Deprecated, to be removed during the
 # K release, at which point the check will be mandatory.
@@ -142,7 +145,7 @@
 # =========== items for agent management extension =============
 # Seconds to regard the agent as down; should be at least twice
 # report_interval, to be sure the agent is down for good
-# agent_down_time = 75
+agent_down_time = 75
 # ===========  end of items for agent management extension =====

 # =========== items for agent scheduler extension =============
@@ -156,11 +159,11 @@
 # Allow auto scheduling networks to DHCP agent. It will schedule non-hosted
 # networks to first DHCP agent which sends get_active_networks message to
 # neutron server
-# network_auto_schedule = True
+network_auto_schedule = True

 # Allow auto scheduling routers to L3 agent. It will schedule non-hosted
 # routers to first L3 agent which sends sync_routers message to neutron server
-# router_auto_schedule = True
+router_auto_schedule = True

 # Allow automatic rescheduling of routers from dead L3 agents with
 # admin_state_up set to True to alive agents.
@@ -267,6 +270,9 @@
 # send_events_interval = 2

 # ======== end of neutron nova interactions ==========
+use_syslog_rfc_format=true
+kombu_reconnect_delay=5.0
+firewall_driver=neutron.agent.linux.iptables_firewall.OVSHybridIptablesFirewallDriver

 #
 # Options defined in oslo.messaging
@@ -348,21 +354,22 @@

 # RabbitMQ HA cluster host:port pairs. (list value)
 #rabbit_hosts=$rabbit_host:$rabbit_port
+rabbit_hosts = %RABBIT_HOSTS%

 # Connect over SSL for RabbitMQ. (boolean value)
 #rabbit_use_ssl=false

 # The RabbitMQ userid. (string value)
-#rabbit_userid=guest
+rabbit_userid = %RABBIT_USERID%

 # The RabbitMQ password. (string value)
-#rabbit_password=guest
+rabbit_password=%RABBIT_PASSWORD%

 # the RabbitMQ login method (string value)
 #rabbit_login_method=AMQPLAIN

 # The RabbitMQ virtual host. (string value)
-#rabbit_virtual_host=/
+rabbit_virtual_host=/

 # How frequently to retry connecting with RabbitMQ. (integer
 # value)
@@ -379,8 +386,8 @@
 # Use HA queues in RabbitMQ (x-ha-policy: all). If you change
 # this option, you must wipe the RabbitMQ database. (boolean
 # value)
-#rabbit_ha_queues=false
-
+rabbit_ha_queues= True
+
 # If passed, use a fake RabbitMQ provider. (boolean value)
 #fake_rabbit=false

@@ -425,7 +432,7 @@

 # Driver or drivers to handle sending notifications. (multi
 # valued)
-#notification_driver=
+notification_driver=neutron.openstack.common.notifier.rpc_notifier

 # AMQP topic used for OpenStack notifications. (list value)
 # Deprecated group/name - [rpc_notifier2]/topics
@@ -441,12 +448,12 @@

 # The messaging driver to use, defaults to rabbit. Other
 # drivers include qpid and zmq. (string value)
-#rpc_backend=rabbit
+rpc_backend=neutron.openstack.common.rpc.impl_kombu

 # The default exchange under which topics are scoped. May be
 # overridden by an exchange name specified in the
 # transport_url option. (string value)
-#control_exchange=openstack
+control_exchange=neutron


 [matchmaker_redis]
@@ -549,16 +556,19 @@
 # seconds between nodes reporting state to server; should be less than
 # agent_down_time, best if it is half or less than agent_down_time
 # report_interval = 30
+report_interval = 5

 # ===========  end of items for agent management extension =====

 [keystone_authtoken]
-auth_host = 127.0.0.1
+auth_host = %AUTH_HOST%
 auth_port = 35357
 auth_protocol = http
-admin_tenant_name = %SERVICE_TENANT_NAME%
-admin_user = %SERVICE_USER%
-admin_password = %SERVICE_PASSWORD%
+admin_tenant_name = %ADMIN_TENANT_NAME%
+admin_user = %ADMIN_USER%
+admin_password = %ADMIN_PASSWORD%
+signing_dir = $state_path/keystone-signing
+auth_url=%AUTH_URL%

 [database]
 # This line MUST be changed to actually run the plugin.
diff -Nur neutron/plugins/ml2/ml2_conf.ini /etc/neutron/plugins/ml2/ml2_conf.ini
--- neutron/plugins/ml2/ml2_conf.ini	2014-12-16 15:09:27.000000000 +0000
+++ /etc/neutron/plugins/ml2/ml2_conf.ini	2015-03-12 11:17:08.089823591 +0000
@@ -4,6 +4,7 @@
 #
 # type_drivers = local,flat,vlan,gre,vxlan
 # Example: type_drivers = flat,vlan,gre,vxlan
+type_drivers = %TYPE_DRIVERS%

 # (ListOpt) Ordered list of network_types to allocate as tenant
 # networks. The default value 'local' is useful for single-box testing
@@ -11,6 +12,7 @@
 #
 # tenant_network_types = local
 # Example: tenant_network_types = vlan,gre,vxlan
+tenant_network_types = %TENANT_NETWORK_TYPES%

 # (ListOpt) Ordered list of networking mechanism driver entrypoints
 # to be loaded from the neutron.ml2.mechanism_drivers namespace.
@@ -20,6 +22,7 @@
 # Example: mechanism_drivers = cisco,logger
 # Example: mechanism_drivers = openvswitch,brocade
 # Example: mechanism_drivers = linuxbridge,brocade
+mechanism_drivers = openvswitch

 # (ListOpt) Ordered list of extension driver entrypoints
 # to be loaded from the neutron.ml2.extension_drivers namespace.
@@ -32,6 +35,7 @@
 # physical_network names.
 #
 # flat_networks =
+flat_networks = %FLAT_NETWORKS%
 # Example:flat_networks = physnet1,physnet2
 # Example:flat_networks = *

@@ -43,10 +47,12 @@
 #
 # network_vlan_ranges =
 # Example: network_vlan_ranges = physnet1:1000:2999,physnet2
+network_vlan_ranges = %NETWORK_VLAN_RANGES%

 [ml2_type_gre]
 # (ListOpt) Comma-separated list of <tun_min>:<tun_max> tuples enumerating ranges of GRE tunnel IDs that are available for tenant network allocation
 # tunnel_id_ranges =
+firewall_driver=neutron.agent.linux.iptables_firewall.OVSHybridIptablesFirewallDriver

 [ml2_type_vxlan]
 # (ListOpt) Comma-separated list of <vni_min>:<vni_max> tuples enumerating
@@ -69,3 +75,13 @@
 # Use ipset to speed-up the iptables security groups. Enabling ipset support
 # requires that ipset is installed on L2 agent node.
 # enable_ipset = True
+
+[agent]
+l2_population=False
+polling_interval=2
+arp_responder=False
+
+[ovs]
+enable_tunneling=False
+integration_bridge=br-int
+bridge_mappings=physnet2:br-prv
