--- a/ceph/manifests/conf.pp
+++ b/ceph/manifests/conf.pp
@@ -7,7 +7,7 @@
   if $node_hostname == $::ceph::primary_mon {

     exec {'ceph-deploy new':
-      command   => "ceph-deploy new ${node_hostname}:${mon_addr}",
+      command   => "ceph-deploy --overwrite-conf new ${node_hostname}:${mon_addr}",
       cwd       => '/etc/ceph',
       logoutput => true,
       creates   => '/etc/ceph/ceph.conf',
