diff --git a/galera/manifests/init.pp b/galera/manifests/init.pp
index 7f25ab3..8fe8633 100644
--- a/galera/manifests/init.pp
+++ b/galera/manifests/init.pp
@@ -76,6 +76,11 @@
 # size to exceed the value of this variable, the server rotates the
 # binary logs (closes the current file and opens the next one). The
 # minimum value is 4096 bytes. The maximum and default value is 512MB.
+#
+# [*ignore_db_dirs*]
+#  (optional) array of directories to ignore in datadir.
+#  Defaults to []
+#

 class galera (
   $cluster_name         = 'openstack',
@@ -95,6 +100,7 @@ class galera (
   $binary_logs_enabled  = false,
   $binary_logs_period   = 1,
   $binary_logs_maxsize  = '512M',
+  $ignore_db_dirs       = [],
   ) {

   include galera::params
diff --git a/galera/templates/my.cnf.erb b/galera/templates/my.cnf.erb
index 253a8a2..e832d2c 100644
--- a/galera/templates/my.cnf.erb
+++ b/galera/templates/my.cnf.erb
@@ -1,10 +1,10 @@
-# #[mysqld]
-# #datadir=/var/lib/mysql
-# #socket=/var/lib/mysql/mysql.sock
-# #user=mysql
-# # Disabling symbolic-links is recommended to prevent assorted security risks
-# #symbolic-links=0
+<% if @ignore_db_dirs and (@ignore_db_dirs.length > 0) -%>
+[mysqld]
+  <%- @ignore_db_dirs.each do |directory| -%>
+ignore-db-dir=<%= directory %>
+  <%- end -%>

+<% end -%>
 [mysqld_safe]
 <% if @use_syslog -%>
 syslog
@@ -12,6 +12,4 @@ syslog
 log-error=/var/log/mysqld.log
 <% end -%>

-# pid-file=/var/run/mysqld.pid
-
 !includedir /etc/mysql/conf.d/
diff --git a/mysql/manifests/server.pp b/mysql/manifests/server.pp
index 4cce092..a9f0ae9 100644
--- a/mysql/manifests/server.pp
+++ b/mysql/manifests/server.pp
@@ -227,6 +227,7 @@ class mysql::server (
       skip_name_resolve  => $mysql_skip_name_resolve,
       use_syslog         => $use_syslog,
       wsrep_sst_password => $root_password,
+      ignore_db_dirs     => $ignore_db_dirs,
     }

   }
