diff -Nura a/usr/lib/python2.6/site-packages/nailgun/api/v1/urls.py b/usr/lib/python2.6/site-packages/nailgun/api/v1/urls.py
--- a/usr/lib/python2.6/site-packages/nailgun/api/v1/urls.py	2015-06-05 11:32:58.786062277 +0000
+++ b/usr/lib/python2.6/site-packages/nailgun/api/v1/urls.py	2015-06-05 11:30:49.053210286 +0000
@@ -264,6 +264,10 @@
     MasterNodeSettingsHandler,
 )
 
+import octane.urls
+urls += octane.urls.urls
+from octane.handlers import *
+
 urls = [i if isinstance(i, str) else i.__name__ for i in urls]
 
 _locals = locals()
