diff --git a/mcagents/erase_node.rb b/mcagents/erase_node.rb
index a2f0565..68b07ee 100644
--- a/mcagents/erase_node.rb
+++ b/mcagents/erase_node.rb
@@ -106,12 +106,6 @@ module MCollective
             end
 
             begin
-              get_boot_devices.each do |dev|
-                erase_partitions(dev[:name])
-                erase_data(dev[:name])
-                erase_data(dev[:name], 1, dev[:size], '512')
-              end
-
               reply[:erased] = true
             rescue Exception => e
               reply[:erased] = false
