diff --git a/pmanager.py b/pmanager.py
index 2c081d8..78013df 100644
--- a/pmanager.py
+++ b/pmanager.py
@@ -696,8 +696,26 @@ class PreseedPManager(object):
     def num_ceph_osds(self):
         return self.get_partition_count('ceph')
 
+    def contains_save_ceph(self, disk=None, part=None):
+        def is_save_partition(part):
+            return (part.get("name") == "ceph" and
+                        part["type"] == "partition" and
+                        part["size"] > 0 and
+                        part.get("keep"))
+        if disk is not None:
+            for volume in disk["volumes"]:
+                if is_save_partition(volume):
+                    return True
+            return False
+        elif part is not None:
+            return is_save_partition(part)
+
     def erase_partition_table(self):
         for disk in self.iterdisks():
+            if self.contains_save_ceph(disk=disk):
+                self.early("parted -s {0} print free"
+                           .format(self._disk_dev(disk)))
+                continue
             self.early("test -e {0} && "
                        "dd if=/dev/zero of={0} "
                        "bs=1M count=10".format(self._disk_dev(disk)))
@@ -913,6 +931,9 @@ class PreseedPManager(object):
                 self.late("cat /proc/mdstat")
                 self.late("cat /proc/partitions")
 
+                if self.contains_save_ceph(part=part):
+                    continue
+
                 # clear any fs info that may remain on newly created partition
                 self.late("dd if=/dev/zero of={0}{1}{2} bs=1M count=10"
                           "".format(self._disk_dev(disk),
